maestrowf/datastructures/core/study.py

```diff
@@ -127,6 +130,48 @@ def real_name(self):
         """
         return self._name
 
+    @property
+    def weight(self):
+        """
+        Get the weight for this step.
+
+        :returns: int.
+        """
+        return self._weight
+
+    @weight.setter
+    def weight(self, new_weight):
+        """
+        Set the weight of the step.
+
+        :param new_weight: integer representing the weight of this step.
+        """
+        self._weight = new_weight
+
+    def __lt__(self, other):
+        """
+        Less than operator for the StudyStep class.
+
+        :param other: Object to compare self to.
+        :returns: True if weight is less than other's weight
+        """
+        if self.weight < other.weight:
+            return True
+
+        return False
```

Comment:
To add onto this, I think you're only required to implement `__lt__` and `__eq__` since you can build the other comparison methods from them. But I'm only 85% sure from memory 😅

Reply:
I'll throw another wrench into this and point out this implementation is probably not what we want despite it working for the current feature.  I.e. step weights can't be the only basis of comparison.  The `__eq__` gets used for doing step equality, which means we can't use `__lt__` to construct that if we stick with this `__lt__` that only cares about the step weights.    Had kind of punted on what other things to care about in these during the initial implementation, but should definitely be addressed now